host/xtest/acipher_perf.c

```diff
@@ -0,0 +1,942 @@
+			op.params[2].memref.parent = &in_shm;
+			op.params[2].memref.size = size;
+			op.params[3].memref.parent = &out_shm;
+			op.params[3].memref.size = out_shm.size;
+			break;
+		case MODE_SIGN:
+			op.params[2].memref.parent = &hash_shm;
+			op.params[2].memref.size = hash_shm.size;
+			op.params[3].memref.parent = &out_shm;
+			op.params[3].memref.size = out_shm.size;
+			break;
+		case MODE_DECRYPT:
+			prepare_enc_sign(size, mode, is_random, buf, blen);
+			op.params[2].memref.parent = &out_shm;
+			op.params[2].memref.size = out_shm.size;
+			op.params[3].memref.parent = &in_shm;
+			op.params[3].memref.size = size;
+			break;
+		case MODE_VERIFY:
+			prepare_enc_sign(size, mode, is_random, buf, blen);
+			op.params[2].memref.parent = &hash_shm;
+			op.params[2].memref.size = hash_shm.size;
+			op.params[3].memref.parent = &out_shm;
+			op.params[3].memref.size = out_shm.size;
+			break;
+		default:
+			fprintf(stderr, "Unexpected mode value\n");
+			exit(1);
+		}
+	}
+
+	if (warmup)
+		do_warmup(warmup);
+
+	while (n-- > 0) {
+		if (mode == MODE_ENCRYPT && main_algo == RSA) {
+			/* Make sure the N > M */
+			((unsigned char *)(in_shm.buffer))[0] = 0x00;
+			/* Avoid the problem that the last encryption result is
+			shorter than the plaintext. */
+			op.params[3].memref.size = out_shm.size;
+		}
+
+		get_current_time(&t0);
+
+		res = TEEC_InvokeCommand(&sess, cmd, &op, &ret_origin);
+
+		check_res(res, "TEEC_InvokeCommand()", &ret_origin);
+
+		get_current_time(&t1);
+
+		time += 1.0f * timespec_diff_ns(&t0, &t1) / 1000.0f / 1000.0f / 1000.0f;
+
+		update_stats(&stats, timespec_diff_ns(&t0, &t1));
+		if (n % (n0 / 10) == 0)
+			vverbose("#");
+	}
+
+	vverbose("\n");
+	sd = stddev(&stats);
+	printf("Ops=%.3f kops\n", n * l / time / 1000.0f);
+	printf("min=%gus max=%gus mean=%gus stddev=%gus (cv %g%%) (%gMiB/s)\n",
+		stats.min / 1000, stats.max / 1000, stats.m / 1000,
+		sd / 1000, 100 * sd / stats.m, mb_per_sec(size, stats.m));
+	verbose("2-sigma interval: %g..%gus (%g..%gMiB/s)\n",
+		(stats.m - 2 * sd) / 1000, (stats.m + 2 * sd) / 1000,
+		mb_per_sec(size, stats.m + 2 * sd),
+		mb_per_sec(size, stats.m - 2 * sd));
+
+	free_shm();
+	free(buf);
+	close_ta();
+	return 0;
+}
+
+#define NEXT_ARG(i) \
+	do { \
+		if (++i == argc) { \
+			fprintf(stderr, "%s: %s: missing argument\n", \
+				argv[0], argv[i - 1]); \
+			return 1; \
+		} \
+	} while (0);
+
+#define USAGE() usage(argv[0], width_bits, main_algo, mode, \
+		      salt_len, size, rsa_algo, warmup, l, n)
+
+static int get_cipher_hash_len(uint32_t algo)
+{
+	switch (algo) {
+	case RSAES_PKCS1_OAEP_SHA1:
+		return SHA1_LEN;
+	case RSAES_PKCS1_OAEP_SHA224:
+		return SHA224_LEN;
+	case RSAES_PKCS1_OAEP_SHA256:
+		return SHA256_LEN;
+	case RSAES_PKCS1_OAEP_SHA384:
+		return SHA384_LEN;
+	case RSAES_PKCS1_OAEP_SHA512:
+		return SHA512_LEN;
+	default:
+		fprintf(stderr, "The algo[%u] is not valid!\n", algo);
+	}
+
+	return -1;
+}
+
+static int get_sign_hash_len(uint32_t algo)
```

Comment:
You can combine this function with the one above except if you want to use there function as a safety check

Reply:
solved